src/ol/source/Tile.js

```diff
@@ -281,8 +281,9 @@ class TileSource extends Source {
    * @protected
    */
   getTileCacheForProjection(projection) {
+    const sourceProjection = this.getProjection();
```

Comment:
While at it, you could add `|null` as return type for `ol/Source#getProjection()`.